core/src/main/scala/akka/persistence/r2dbc/snapshot/SnapshotDao.scala

```diff
@@ -0,0 +1,278 @@
+    val entityTypeHint = SliceUtils.extractEntityTypeHintFromPersistenceId(persistenceId)
+    val slice = SliceUtils.sliceForPersistenceId(persistenceId, settings.maxNumberOfSlices)
+
+    var paramIdx = 3
+    val selectSnapshots = s"SELECT * FROM $snapshotTable " +
+      s"WHERE slice = $$1 AND entity_type_hint = $$2 AND persistence_id = $$3" +
+      (if (criteria.maxSequenceNr != Long.MaxValue) {
+         paramIdx += 1
+         s" AND sequence_number <= $$$paramIdx"
+       } else "") +
+      (if (criteria.minSequenceNr > 0L) {
+         paramIdx += 1
+         s" AND sequence_number >= $$$paramIdx"
+       } else "") +
+      (if (criteria.maxTimestamp != Long.MaxValue) {
+         paramIdx += 1
+         s" AND write_timestamp <= $$$paramIdx"
+       } else "") +
+      (if (criteria.minTimestamp != 0L) {
+         paramIdx += 1
+         s" AND write_timestamp >= $$$paramIdx"
+       } else "") +
+      " ORDER BY sequence_number DESC"
+
+    r2dbcExecutor
+      .select(s"select snapshot [$persistenceId], criteria: [$criteria]")(
+        { connection =>
+          val statement = connection
+            .createStatement(selectSnapshots)
+            .bind(0, slice)
+            .bind(1, entityTypeHint)
+            .bind(2, persistenceId)
+
+          var bindIdx = 2
+          if (criteria.maxSequenceNr != Long.MaxValue) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.maxSequenceNr)
+          }
+          if (criteria.minSequenceNr > 0L) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.minSequenceNr)
+          }
+          if (criteria.maxTimestamp != Long.MaxValue) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.maxTimestamp)
+          }
+          if (criteria.minTimestamp > 0L) {
+            bindIdx += 1
+            statement.bind(bindIdx, criteria.minTimestamp)
+          }
+          statement
+        },
+        row =>
+          Serialization.withTransportInformation(system.classicSystem.asInstanceOf[ExtendedActorSystem]) { () =>
+            deserializeSnapshotRow(row, serialization)
+          })
+      .map(_.headOption)(ExecutionContexts.parasitic)
+
+  }
+
+  def store(metadata: SnapshotMetadata, value: Any): Future[Unit] =
+    Serialization.withTransportInformation(system.classicSystem.asInstanceOf[ExtendedActorSystem]) { () =>
+      val entityTypeHint = SliceUtils.extractEntityTypeHintFromPersistenceId(metadata.persistenceId)
+      val slice = SliceUtils.sliceForPersistenceId(metadata.persistenceId, settings.maxNumberOfSlices)
+
+      val insert =
+        if (metadata.metadata.isEmpty) insertSql
+        else insertSqlMeta
+
+      val snapshot = value.asInstanceOf[AnyRef]
+      val snapshotSerializer = serialization.findSerializerFor(snapshot)
+      val snapshotManifest = Serializers.manifestFor(snapshotSerializer, snapshot)
+      val serializedSnapshot = snapshotSerializer.toBinary(snapshot)
```

Comment:
in the journal we have
```
          val serialized = serialization.serialize(event).get
          val serializer = serialization.findSerializerFor(event)
          val manifest = Serializers.manifestFor(serializer, event)
          val id: Int = serializer.identifier
```

and then the `Serialization.withTransportInformation` isn't needed

Reply:
Thanks, was wondering why I needed that when it wasn't needed elsewhere.